pkg/client/logs.go

```diff
@@ -180,8 +181,47 @@ type logStreamer struct {
 	client             kubernetes.Interface
 }
 
+func isContainerRunning(statuses *[]v1.ContainerStatus, containerName string) bool {
+	for _, cs := range *statuses {
+		if cs.Name == containerName && cs.State.Running != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func (l *logStreamer) waitForContainerRunning() error {
+	backoffSeconds := 1 * time.Second
+	for {
+		pod, err := l.client.CoreV1().Pods(l.ns).Get(l.pod, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get pod [%s/%s]", l.ns, l.pod)
+		}
+
+		for _, cs := range pod.Status.ContainerStatuses {
+			if cs.Name == l.container && cs.State.Running != nil {
```

Comment:
It looks like you moved all the logic of this loop into the `isContainerRunning(...) method`; just need to call it instead of looping right?

Reply:
yeah.. that's right! good catch